zui/src/z/types/utils/index.ts

```diff
@@ -82,6 +82,17 @@ export namespace util {
     }
     return value
   }
+
+  export const compareFunctions = (a: Function, b: Function) => {
+    /**
+     * The only proper way to deeply compare functions would be to ensure they return the same value for the same input.
+     * This is impossible to do unless the domain of the function is known and the function is pure.
+     *
+     * Comparing source code is not ideal since 2 function could be equivalent but have different source code,
+     * but that's our best option.
+     */
```

Comment:
**praise:** good comment; makes the intent clear and shows that other avenues were explored